### PR TITLE
MORE INFO FOR SJ PREVIEWS: As Emmanuel, I want preview and harvest pages to be updated with ActionCable, so that I am able to see more information about what is going on with SJ

### PR DESCRIPTION
**Pull request by Reinkaos**
**Acceptance Criteria**
- Harvest operators can see timely information about progress and key events when running harvests, enrichments and previews
- UAT SJ stack used to minimise disruption to harvesters

### CODE REVIEW

File: .rubocop_todo.yml [hunk 1]
```diff
@@ -13,9 +13,9 @@ Lint/HandleExceptions:
 
 # Offense count: 19
 Metrics/AbcSize:
-  Max: 56
+  Max: 100
```
Comment (EdwinRozario) [inline .rubocop_todo.yml]: Thats a big difference. Cant you add a Rubocop exception on the specific methods?

File: app/models/preview.rb [hunk 2]
```diff
@@ -32,4 +37,48 @@ def self.spawn_preview_worker(attributes)
...
+    CodeRay.scan(field_errors_json, :json).html(line_numbers: :table).html_safe if field_errors?
+  end
+
+  def harvest_job_errors_output
+    JSON.parse(harvest_job_errors) if harvest_job_errors.present?
+  end
+
+  def harvest_failure_output
+    JSON.parse(harvest_failure) if harvest_failure.present?
+  end
```
Comment (EdwinRozario) [inline app/models/preview.rb]: Specs for these methods?

Comment (Reinkaos) [reply app/models/preview.rb]: Oh yeah sure thing, forgot to copy them from the manager. Good spotting!

File: config/cable.yml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+#   adapter: async
+#
+# production:
+#   adapter: redis
+#   ur: redis-url
+#   channel_prefix: manager_production
+
+redis: &redis
+  adapter: redis
+  url: redis://localhost:6379/3
```
Comment (tim-boost) [inline config/cable.yml]: Does this need to be configurable for the cloud?

File: spec/workers/preview_worker_spec.rb [hunk 4]
```diff
@@ -4,7 +4,8 @@
 describe PreviewWorker do
   let(:parser) { Parser.new(strategy: 'xml', name: 'Natlib Pages', content: 'class NatlibPages < SupplejackCommon::Xml::Base; end', file_name: 'natlib_pages.rb', source: { source_id: 'source_id' }) }
   let(:job) { HarvestJob.new(environment: 'preview', parser_id: 'abc123', index: 3, harvest_failure: {}, last_posted_record_id: 1234) }
-  let(:preview) { mock_model(Preview, _id: '123').as_null_object }
+  # let(:preview) { mock_model(Preview, _id: '123').as_null_object }
```
Comment (tim-boost) [inline spec/workers/preview_worker_spec.rb]: Can we remove this line?

File: config/cable.yml (hunk 3 above)
Comment (Reinkaos) [reply config/cable.yml]: Yeah I'll make it an environment variable.

File: spec/workers/preview_worker_spec.rb (hunk 4 above)
Comment (Reinkaos) [reply spec/workers/preview_worker_spec.rb]: Yup!

**Comment by Reinkaos** [comment]
you are reborn!

**Pull request closed without merging**